### PR TITLE
Emag "charges and recharges"

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
Emags now have 4 charges that recharge over time flavored as it recalibrating. I have added an admin version known as the elite cryptographic sequencer with a practically infinite amount of charges.

## Why It's Good For The Game
Personally I don't think any changes are necessary but I saw that people thought that this would make a better alternative and my morning was free. This will probably be closed very soon.

## Changelog
:cl: itseasytosee
tweak: using emags a lot within a short period of time will require the emag to take a second to recalibrate before it may be used again
/:cl:

### CODE REVIEW

**Comment by SumFaggotPlayTester** [comment]
This seems like a good balance.

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -61,21 +61,46 @@
...
+	var/charge_tick = 0
 
 /obj/item/card/emag/bluespace
 	name = "bluespace cryptographic sequencer"
 	desc = "It's a blue card with a magnetic strip attached to some circuitry. It appears to have some sort of transmitter attached to it."
 	color = rgb(40, 130, 255)
 	prox_check = FALSE
 
+/obj/item/card/emag/elite
+	name = "advaced cryptographic sequencer"
```
Comment (Angustmeta) [inline code/game/objects/items/cards_ids.dm:83]:
```suggestion
	name = "advanced cryptographic sequencer"

```

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -61,21 +61,47 @@
...
 
 /obj/item/card/emag/bluespace
 	name = "bluespace cryptographic sequencer"
 	desc = "It's a blue card with a magnetic strip attached to some circuitry. It appears to have some sort of transmitter attached to it."
 	color = rgb(40, 130, 255)
 	prox_check = FALSE
 
+/obj/item/card/emag/elite
+	name = "advanced cryptographic sequencer"
+
```
Comment (Angustmeta) [inline code/game/objects/items/cards_ids.dm:84]:
```suggestion
```

Comment (Angustmeta) [reply code/game/objects/items/cards_ids.dm:84]: My bad

**Comment by ATH1909** [comment]
Suggestion: Add a way to examine your card to see how many charges you currently have left on it and how long it'll take to recharge.

Check out how Fulp did it in these PRs:
TheSwain/Fulpstation#273
TheSwain/Fulpstation#295

**Comment by ATH1909** [comment]
For reference, I think that emags are fine and don't need a nerf (especially since no one can agree on WHY they "need" a nerf, with some of the reasons given directly conflicting with each other), but hey, someone's whipped up the coders into a frenzy about how emags "have" to be nerfed, and I can't exactly stop that.

**Comment by itseasytosee** [comment]
First off I also do not want an emag nerf and think it's fine, but as you say people are now in a nerf frenzy. Also, I don't know how I would flavor the recharges because they aren't really "charges" and I think making it such would not make sense as electromagnetic is literally in the name. Maybe I'll make a calibration diode that glows a different color or something on examination depending on charges.
>3 the diode is green, 1-3 the diode is yellow, 0 diode is red.

**Comment by Shadowflame909** [comment]
This nerf isn't necessary and I think your counter-nerf to try and stiffen the blow of losing a very valuable and interesting traitor item should be put on hold until we gage how the maintainers actually feel about goofs initial PR.

It's like someone gave a suggestion that we should drive the bus off the cliff, and you start inflating a giant airbag under the cliff to save the people inside the bus, but you don't know if the bus driver agrees with that suggestion.

They might just because of the fact that there is an air-bag under that cliff now!

**Comment by itseasytosee** [comment]
Solid analogy. The plan was to close it after I got some opinions and have a functional alternative that people could use.

**Comment by wesoda25** [comment]
ok but if they are serious about merging one of those this better get reopened bc its way better

**Comment by itseasytosee** [comment]
thats the intent

**Comment by MrDoomBringer** [comment]
@itseasytosee whats your discord goddammit

**Comment by itseasytosee** [comment]
Zirconium Eyes#2424

**Pull request closed without merging**